lib/stdlib/src/uri_string.erl

```diff
@@ -375,6 +375,80 @@ recompose(Map) ->
     end.
 
 
+%%-------------------------------------------------------------------------
+%% Resolve URIs
+%%-------------------------------------------------------------------------
+-spec resolve(RefURI, BaseURI) -> TargetURI when
+      RefURI :: uri_string() | uri_map(),
+      BaseURI :: uri_string() | uri_map(),
+      TargetURI :: uri_string()
+                 | error().
+resolve(URIMap, BaseURIMap) ->
+    resolve(URIMap, BaseURIMap, []).
+
+
+-spec resolve(RefURI, BaseURI, Options) -> TargetURI when
+      RefURI :: uri_string() | uri_map(),
+      BaseURI :: uri_string() | uri_map(),
+      Options :: [return_map],
+      TargetURI :: uri_string() | uri_map()
+                 | error().
+resolve(URIMap, BaseURIMap, Options) ->
+    try resolve1(URIMap, BaseURIMap, Options)
+    catch
+        throw:{error, Atom, RestData} -> {error, Atom, RestData}
+    end.
+
+resolve1(URI, BaseURI, Options) ->
```

Comment:
I would move helper functions to "Internal functions" to keep them separated from the API.

Reply:
Yes I guessed I would have to. I will do that when I update the documentation.